apps/ws/src/SocketManager.ts

```diff
@@ -17,10 +17,12 @@ export class SocketManager {
   private static instance: SocketManager;
   private interestedSockets: Map<string, User[]>;
   private userRoomMappping: Map<string, string>;
+  private spectatorSockets: Map<string, User[]>;
```

Comment:
Is there a need for saparat e`spectatorSockets`?

Reply:
They might be waiting for Playing. so added new Map of Sepectator